test/libyul/yulOptimizerTests/fullSSAReverse/store.yul

```diff
@@ -0,0 +1,13 @@
+{
+    phi_store("a", 42)
+    let b := 0
+    phi_store("b", 23)
+}
```

Comment:
```suggestion
{
    phi_store("a", 42)
    let a := 20
    let b := 0
    phi_store("b", 23)
}
```
Don't we have to do some error checks for cases like this?

Reply:
Probably, yeah. I was about to say that this can't happen, but maybe it *could* happen, if the original ``a`` (or let it be ``a_2``) becomes unused and removed entirely and then the name dispenser creates a new variable with the same name - I'll need to think about it. We might also just avoid that by reserving those names (in the first arguments of ``phi_store``) in the name dispenser as well, not sure.